project/src/main/java/com/google/sps/servlets/CreateBusinessAccountServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps.servlets;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+@WebServlet("/createBusinessAccount")
+public class CreateBusinessAccountServlet extends HttpServlet {
+
+  protected DatastoreService datastore;
+  protected UserService userService;
+
+  public CreateBusinessAccountServlet() {
+    super();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    userService = UserServiceFactory.getUserService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get required user information.
+    String userId = userService.getCurrentUser().getUserId();
```

Comment:
Is there any chance of the current user not being created/initialized yet? Does something different need to happen in that case?

Reply:
Before I route to the createBusinessAccount page, I first direct the user to a login servlet that ensures that the user is logged in.